test/framework/backuppolicy/backuppolicy.go

```diff
@@ -0,0 +1,182 @@
+package backuppolicy
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/anynines/a8s-backup-manager/api/v1beta3"
+	"github.com/anynines/a8s-deployment/test/framework"
+	. "github.com/onsi/gomega"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	runtimeClient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	// asyncOpsTimeoutMins is the amount of minutes after which assertions fail if the condition
+	// they check has not become true. Needed because some conditions might become true only
+	// after some time, so we need to check them asynchronously.
+	// TODO: Make asyncOpsTimeoutMins an invocation parameter.
+	asyncOpsTimeoutMins = time.Minute * 2
+	suffixLength        = 6
+	pollingPeriod       = 1 * time.Second
+)
+
+// Option represents a functional option for backup objects. To learn what a functional option is,
+// read here: https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis
+type Option func(*v1beta3.BackupPolicy)
+
+func SetNamespacedName(namespace string) Option {
+	return func(p *v1beta3.BackupPolicy) {
+		p.Name = framework.UniqueName("backup-policy", suffixLength)
+		p.Namespace = namespace
+	}
+}
+
+func SetInstanceRef(dsi runtimeClient.Object) Option {
+	return func(p *v1beta3.BackupPolicy) {
+		p.Spec.ServiceInstance.APIGroup = dsi.GetObjectKind().GroupVersionKind().Group
+		p.Spec.ServiceInstance.Kind = dsi.GetObjectKind().GroupVersionKind().Kind
+		p.Spec.ServiceInstance.Name = dsi.GetName()
+	}
+}
+
+func SetEnabled(enabled bool) Option {
+	return func(p *v1beta3.BackupPolicy) {
+		p.Spec.Enabled = enabled
+	}
+}
+
+func ScheduleConfiguration(scheduleConfig v1beta3.ScheduleConfiguration) Option {
+	return func(p *v1beta3.BackupPolicy) {
+		p.Spec.ScheduleConfig = scheduleConfig
+	}
+}
+
+func RetentionConfiguration(retentionConfig *v1beta3.RetentionConfiguration) Option {
+	return func(p *v1beta3.BackupPolicy) {
+		p.Spec.RetentionConfig = retentionConfig
+	}
+}
+
+func New(opts ...Option) *v1beta3.BackupPolicy {
+	p := &v1beta3.BackupPolicy{}
+	for _, opt := range opts {
+		opt(p)
+	}
+	return p
+}
+
+func GetExistingBackups(p *v1beta3.BackupPolicy, dsi runtimeClient.Object) []v1beta3.Backup {
```

Comment:
```suggestion
func MockExistingBackups(p *v1beta3.BackupPolicy, dsi runtimeClient.Object) []v1beta3.Backup {
```
Are `existingBackups` test data?